cmd/notation/policy/show.go

```diff
@@ -0,0 +1,50 @@
+package policy
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/notaryproject/notation-go/dir"
+	"github.com/notaryproject/notation/internal/cmd"
+	"github.com/spf13/cobra"
+)
+
+type showOpts struct {
+	cmd.LoggingFlagOpts
+}
+
+func showCmd() *cobra.Command {
+	var opts showOpts
+	command := &cobra.Command{
+		Use:   "show [flags]",
+		Short: "[Preview] Show trust policy configuration",
+		Long: `[Preview] Show trust policy configuration.
+
+** This command is in preview and under development. **
+
+Example - Show current trust policy configuration:
+  notation policy show
+
+Example - Save current trust policy configuration to a file:
+  notation policy show > my_policy.json
+`,
+		Args: cobra.ExactArgs(0),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return runShow(cmd, opts)
+		},
+	}
+	opts.LoggingFlagOpts.ApplyFlags(command.Flags())
+	return command
+}
+
+func runShow(command *cobra.Command, opts showOpts) error {
+	// read policy json contents
+	policyJson, err := os.ReadFile(filepath.Join(dir.UserConfigDir, dir.PathTrustPolicy))
+	if err != nil {
+		fmt.Fprintln(os.Stderr, "no trust policy configuration found, you may import one via `notation policy import <path-to-policy.json>`")
+		return fmt.Errorf("failed to read trust policy configuration: %w", err)
+	}
```

Comment:
```suggestion
	policyDoc, err := trustpolicy.LoadDocument()
	if err != nil {
		fmt.Fprintln(os.Stderr, "failed to load trust policy configuration, you may import one via `notation policy import <path-to-policy.json>`")
		return err
	}
	if err := policyDoc.Validate(); err != nil {
		fmt.Fprintln(os.Stderr, "existing trust policy configuration is invalid, you may import one via `notation policy import <path-to-policy.json>`")
		return err
	}
```
we need to call `Validate()` according to our spec: https://github.com/notaryproject/notation/blob/main/specs/commandline/policy.md#usage, i.e., only valid trust policy document shall be returned to the user.

Reply:
I do see the validation for `import` but doesn't find it in `show` @yizha1 Can you help confirm the UX?

Reply:
If that's not in the spec, I think that's something we want to add into it. Because returning an invalid trust policy document to the user might be confusing and not useful.

Reply:
Yes, I agree that validation need to be done. But how to deal with invalid policy is a tricky work and the UX need to be clarified:

Option 1: show validation error only.
Option 2: show validation error and the current configuration.
Option 3: show validation error and the current configuration with invalid part highlighted.

Reply:
IMO,
Option 3 might be the best option, it tells the user which exact part needs to be fixed.
Option 1 is also good enough, as it tells the user the existing trust policy is malformed, import a new one to replace it. (Note, we do not have a `notation policy init` command, this means that we can fairly assume at this point user knows all the required specs of a notation trust policy document.)
Option 2 is not what we want. Since it's an invalid configuration, merely showing it does not add any value to the user.

Reply:
> I do see the validation for `import` but doesn't find it in `show` @yizha1 Can you help confirm the UX?

Why we need validation for show command? Show command just displays the current trust policy configuration which should be valid

Reply:
> > I do see the validation for `import` but doesn't find it in `show` @yizha1 Can you help confirm the UX?
> 
> Why we need validation for show command? Show command just displays the current trust policy configuration which should be valid

trustpolicy.json is not guaranteed to be valid at any point of time, since it's just a file, it can still be manually removed/added by the user without using notation's command. (Btw, that's one of the reasons that we should have a `notation policy init` command.) Think about a new user, they might just compose their own trustpolicy.json then drag and drop it into notation's directory without using `notation policy import`.

Reply:
I chose option 3, but the validation result only shows the invalid part not the whole configuration. @qweeah @patrickzheng200

Reply:
> I chose option 3, but the validation result only shows the invalid part not the whole configuration. @qweeah @patrickzheng200

Yeah, agree. Our `validate()` process fails at the first error it encounters, so it won't show all the failures in the doc. In this case, we could just `print out the error + the certain line of content that triggered this error`.

Reply:
I had an implementation, but the output information may not be accurate, e.g. if the version number doesn't exist, the error message will still mention the first policy.  So to fully support option3, Changes need to be made to [Document.Validate](https://pkg.go.dev/github.com/notaryproject/notation-go@v1.0.0-rc.3/verifier/trustpolicy#Document.Validate) to provide more contextual info.

1. Version validation failure returns certain type of exported errors
2. Separate version and policy validation into two exported functions

Reply:
@qweeah Based on what I discussed with @yizha1, we could just show the error for now, since the error message contains which trust policy statement triggered the error on which field due to what reason. We could improve the logic to show all the errors in the file after RC.4.

Reply:
sure, implemented as suggested